src/components/moderation/modmail.ts

```diff
@@ -32,13 +33,24 @@ function create_embed(title: string, msg: string) {
 export default class Modmail extends BotComponent {
     // Spam prevention, user is added to the timeout set when clicking the modmail_continue button,
     readonly timeout_set = new Set<string>();
-    modmail_id_counter = -1;
 
     readonly monke_set = new SelfClearingMap<Discord.Snowflake, number>(HOUR, HOUR);
 
-    override async on_ready() {
-        const singleton = await this.wheatley.database.get_bot_singleton();
-        this.modmail_id_counter = singleton.modmail_id_counter;
+    database = this.wheatley.database.create_proxy<{
+        component_state: moderation_state;
+    }>();
+
+    async increment_modmail_id() {
+        const res = await this.database.component_state.findOneAndUpdate(
+            { id: "moderation" },
+            {
+                $inc: {
+                    modmail_id: 1,
+                },
+            },
+            { upsert: true, returnDocument: "after" },
```

Comment:
I think we might want `"before"` here for current modmail id logic but that might behave weirdly if the upsert is something you'd like to rely on

Reply:
Yeah, that's actually what I initially did, but that then introduces the need to handle the special case of there not having been a previous entry, so this turned out to be simpler.